lib/param_parsing/toml.go

```diff
@@ -27,7 +28,7 @@ func getParamsTOML(params Params) (Params, error) {
 		}
 
 		if viperParser.Get("bin") != nil {
-			params.CustomBinaryPath = viperParser.GetString("bin")
+			params.CustomBinaryPath = os.ExpandEnv(viperParser.GetString("bin"))
```

Comment:
Why do we need to do this again?

Reply:
Env var expansion has stopped working at some point between `v1.1.1` and current `master`. I couldn't figure out why or what affected it. So with "again" I mean to bring it back:
```shell
> head -1 ~/.tfswitch.toml
bin = "$HOME/bin/terraform"

# v1.1.1
> tfswitch --version
Version: v1.1.1
> tfswitch 1.4.7
03:03:04.106 INFO Installing terraform at "/home/giermulnik/bin"
03:03:04.106 INFO Switched terraform to version "1.4.7"

# master branch
> go run ./main.go 1.4.7
03:00:58.624 INFO Reading configuration from "/home/giermulnik/.tfswitch.toml"
03:00:58.624 DEBUG [install.go:238,InstallProductVersion] Install version 1.4.7. Dry run: false
03:00:58.624 FATAL [install.go:185,installableBinLocation] Binary path ("$HOME/bin/terraform") does not exist. Manually create bin directory "$HOME/bin" and try again.
exit status 1

# fix/allow_env_vars_in_bin_toml_parameter branch
> go run ./main.go 1.4.7
03:01:26.175 INFO Reading configuration from "/home/giermulnik/.tfswitch.toml"
03:01:26.175 DEBUG [install.go:238,InstallProductVersion] Install version 1.4.7. Dry run: false
03:01:26.175 INFO [install.go:249,InstallProductVersion] Switched Terraform to version "1.4.7"
```

Reply:
I see, makes sense.. was this removed? Why did it work before?

Reply:
> I couldn't figure out why or what affected it.

⬆️

Reply:
@warrensbox Please let me know if you're ok with this PR.

Reply:
Thanks for the explanation!